apps/backend/src/slack/buildStatusMessage.ts

```diff
@@ -20,91 +23,108 @@ export function getEventDescription(event: AutomationEvent): string {
+  const buildLabel = getBuildLabel(build.type, status);
+  const branch = compareScreenshotBucket?.branch;
+  const isGithubPrDefined = pullRequest && pullRequest.number;
+  const prUrl = isGithubPrDefined
+    ? `https://github.com/pull/${pullRequest.number}`
+    : null;
 
-  const blocks: SlackMessageBlock[] = [
-    {
-      type: "header",
-      text: {
-        type: "plain_text",
-        text: getEventDescription(event),
-        emoji: true,
-      },
-    },
-    {
-      type: "context",
-      elements: [
-        {
-          type: "mrkdwn",
-          text: `<${buildUrl}|Build #${build.number}: ${build.name}> on *${project.name}*`,
-        },
-      ],
-    },
-    { type: "divider" },
-    {
-      type: "section",
-      fields: [
-        { type: "mrkdwn", text: `*Type:* ${build.type ?? "N/A"}` },
-        { type: "mrkdwn", text: `*Commit:* ${commitShort}` },
-        {
-          type: "mrkdwn",
-          text: `*Branch:* ${compareScreenshotBucket?.branch ?? "N/A"}`,
-        },
-        {
-          type: "mrkdwn",
-          text: `*Screenshots:* ${compareScreenshotBucket?.screenshotCount ?? 0}`,
-        },
-      ],
-    },
-  ];
+  const projectUrl = getRepositoryUrl(project);
+  const branchUrl =
+    branch && projectUrl ? `${projectUrl}/tree/${branch}` : undefined;
+  const commitUrl =
+    commit && projectUrl ? `${projectUrl}/commit/${commit}` : undefined;
 
-  if (pullRequest && pullRequest.number && pullRequest.title) {
-    const prUrl = `https://github.com/pull/${pullRequest.number}`;
-    blocks.push({
-      type: "section",
-      fields: [
-        {
-          type: "mrkdwn",
-          text: `*Pull Request:* <${prUrl}|#${pullRequest.number}: ${pullRequest.title}>`,
-        },
-      ],
-    });
-  }
+  const contextBlock: SlackMessageBlock = {
+    type: "section",
+    text: {
+      type: "mrkdwn",
+      text: `*<${buildUrl}|Build #${build.number} (${build.name})>*`,
+    },
+  };
 
-  blocks.push({
-    type: "actions",
+  const projectBlock: SlackMessageBlock = {
+    type: "context",
     elements: [
       {
-        type: "button",
-        text: {
-          type: "plain_text",
-          text: "View Build",
-        },
-        url: buildUrl,
-        action_id: "view-build-button",
-        style: "primary",
+        type: "mrkdwn",
+        text: `Project: *${project.name}*`,
       },
     ],
-  });
+  };
+
+  const labelBlock: SlackMessageBlock = {
+    type: "section",
+    text: { type: "mrkdwn", text: buildLabel },
+  };
+
+  const detailsBlock: SlackMessageBlock = {
+    type: "section",
+    fields: [
+      statsMessage
+        ? {
+            type: "mrkdwn" as const,
+            text: `*Screenshots:* ${statsMessage}`,
+          }
+        : null,
+      isGithubPrDefined
+        ? {
+            type: "mrkdwn" as const,
+            text: `*PR:* <${prUrl}|#${pullRequest.number}> ${pullRequest.title ?? ""}`,
+          }
+        : null,
```

Comment:
It's a bit bad typing, you should test what you use: `prUrl && pullRequest.number`.